### PR TITLE
feat(mtls): implement of mutual TLS communication for fetching server info.

### PR DESCRIPTION
**Pull request by hetangmodi-crest**
ADDON-70857
Added fetching of the client certificates for mTLS communication for `ServerInfo` class.

### CODE REVIEW

File: solnlib/server_info.py [hunk 1]
```diff
@@ -56,6 +58,28 @@ def __init__(
             port: The port number, default is None.
             context: Other configurations for Splunk rest client.
         """
+        is_localhost: bool = False
```
Comment (artemrys) [inline solnlib/server_info.py:61]:
```suggestion
        is_localhost = False
```

File: solnlib/server_info.py [hunk 2]
```diff
@@ -16,13 +16,15 @@
...
+import os
 import json
 from typing import Any, Dict, Optional
 
+from splunk.rest import getWebCertFile, getWebKeyFile
 from splunklib import binding
-
 from solnlib import splunk_rest_client as rest_client
 from solnlib import utils
+from .splunkenv import get_splunkd_access_info
```
Comment (artemrys) [inline solnlib/server_info.py:27]:
```suggestion
from solnlib.splunkenv import get_splunkd_access_info
```

File: solnlib/server_info.py [hunk 3]
```diff
@@ -56,6 +58,28 @@ def __init__(
...
+            if all([is_localhost, context.get("verify") is None]):
+                # NOTE: this is specifically for mTLS communication
+                # ONLY if scheme, host, port aren't provided AND user hasn't provided server certificate
+                # we set verify to off (similar to 'rest.simpleRequest' implementation)
+                context["verify"] = False
+
+        elif getWebCertFile() is not None:
+            context["cert_file"] = getWebCertFile()
+            if all([is_localhost, context.get("verify") is None]):
+                context["verify"] = False
```
Comment (artemrys) [inline solnlib/server_info.py:81]: Is it possible for this code to have unexpected side effects? This feature is being enabled by Splunk itself, not by the `solnlib`?

**Review by artemrys — CHANGES_REQUESTED** [review]

File: solnlib/server_info.py (hunk 1 above)
Comment (hetangmodi-crest) [reply solnlib/server_info.py]: Made the changes.

File: solnlib/server_info.py (hunk 2 above)
Comment (hetangmodi-crest) [reply solnlib/server_info.py]: Made the changes.

File: solnlib/server_info.py (hunk 3 above)
Comment (hetangmodi-crest) [reply solnlib/server_info.py:81]: No.
If the mTLS is switched on by the user on their Splunk environment, we would by default switch it on too. Moreover, we do give preference to user's configuration, that is, if they want to pass any custom certificate while making an object of the ServerInfo class, we would use that.

**Pull request merged**